tokio-process/src/unix/reap.rs

```diff
@@ -1,17 +1,23 @@
 use super::orphan::{OrphanQueue, Wait};
 use crate::kill::Kill;
-use futures::{Async, Future, Poll, Stream};
+use futures::stream::TryStream;
+use futures::stream::TryStreamExt;
+use std::future::Future;
 use std::io;
 use std::ops::Deref;
+use std::pin::Pin;
 use std::process::ExitStatus;
+use std::task::Context;
+use std::task::Poll;
 
 /// Orchestrates between registering interest for receiving signals when a
 /// child process has exited, and attempting to poll for process completion.
 #[derive(Debug)]
 pub(crate) struct Reaper<W, Q, S>
 where
-    W: Wait,
-    Q: OrphanQueue<W>,
+    W: Wait + Unpin,
```

Comment:
Looking at the interaction between the drop impl and `self.inner_mut()`, it looks like only `W` needs an `Unpin` bound in the declaration. The rest of the `Unpin` bounds can be omitted here and everywhere else (unless a particular impl block needs them)